### PR TITLE
Datapack failure screen

### PR DESCRIPTION
**Pull request by YanisBft**

### CODE REVIEW

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 1]
```diff
@@ -243,5 +247,11 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 		ARG 2 monitor
 	METHOD method_27466 initFont (Z)V
 		ARG 1 forcesUnicode
+	METHOD method_29042 shouldHideMessage (Ljava/util/UUID;)Z
```
Comment (Juuxel) [inline mappings/net/minecraft/client/MinecraftClient.mapping]: What type of messages does this hide?

Comment (YanisBft) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: Chat messages, and check if online chat option is enabled

**Review by modmuss50 — APPROVED** [review]

File: mappings/net/minecraft/client/MinecraftClient.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: maybe `isMuted`?
```suggestion
	METHOD method_29042 isMessageBlocked (Ljava/util/UUID;)Z
```

File: mappings/net/minecraft/util/Util.mapping [hunk 2]
```diff
@@ -3,6 +3,7 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
 	FIELD field_1129 LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD field_18034 NEXT_SERVER_WORKER_ID Ljava/util/concurrent/atomic/AtomicInteger;
 	FIELD field_18035 SERVER_WORKER_EXECUTOR Ljava/util/concurrent/ExecutorService;
+	FIELD field_25140 DEFAULT_UUID Ljava/util/UUID;
```
Comment (liach) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
	FIELD field_25140 NIL_UUID Ljava/util/UUID;
```
https://en.wikipedia.org/wiki/Universally_unique_identifier#Nil_UUID

File: mappings/net/minecraft/client/MinecraftClient.mapping (hunk 1 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: would prefer `shouldHideMessage` or `isChatMuted`

Comment (liach) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: The parameter is already `sender`, so you know if a sender is muted. also this exists for all game messages than just user messages

On another note, this check is used by narrator as well. Checking `shouldHide` in narrator doesn't make as much sense as checking `isMuted`

**Comment by liach** [comment]
Update: changed suggestion to `isMessageBlocked`.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/util/Util.mapping [hunk 3]
```diff
@@ -3,6 +3,7 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
 	FIELD field_1129 LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD field_18034 NEXT_SERVER_WORKER_ID Ljava/util/concurrent/atomic/AtomicInteger;
 	FIELD field_18035 SERVER_WORKER_EXECUTOR Ljava/util/concurrent/ExecutorService;
+	FIELD field_25140 NIL_UUID Ljava/util/UUID;
```
Comment (Earthcomputer) [inline mappings/net/minecraft/util/Util.mapping:6]:
```suggestion
	FIELD field_25140 ZERO_UUID Ljava/util/UUID;
```

Comment (liach) [reply mappings/net/minecraft/util/Util.mapping:6]: https://en.wikipedia.org/wiki/Universally_unique_identifier#Nil_UUID

Comment (Earthcomputer) [reply mappings/net/minecraft/util/Util.mapping:6]: Fair enough

**Review by Earthcomputer — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**